src/wkt/src/message.rs

```diff
@@ -14,12 +14,83 @@
 
 //! Define traits required of all messages.
 
+pub(crate) type Map = serde_json::Map<String, serde_json::Value>;
+use crate::AnyError as Error;
+
 /// A trait that must be implemented by all messages.
 ///
 /// Messages sent to and received from Google Cloud services may be wrapped in
 /// [Any][crate::any::Any]. `Any` uses a `@type` field to encoding the type
 /// name and then validates extraction and insertion against this type.
-pub trait Message {
+pub trait Message: serde::ser::Serialize + serde::de::DeserializeOwned {
```

Comment:
Instead of adding these to the trait can more pub(crate) fns just be added like you did for `to_json_object`?

Reply:
Works for me!